pkg/provisioner/ironic/ironic.go

```diff
@@ -234,6 +234,14 @@ func (p *ironicProvisioner) ValidateManagementAccess(credentialsChanged bool) (r
 	if ironicNode == nil {
 		p.log.Info("registering host in ironic")
 
+		properties := map[string]interface{}{}
+
+		if string(p.host.Spec.BootMode) == "legacy" {
```

Comment:
Is this going to change the current default behavior by requiring users to set the legacy boot mode?

Can we skip setting the boot mode entirely if it is not set in the host spec?

The enum values are define above, why compare to a literal string here?